libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -983,3 +1006,68 @@ eth::AssemblyPointer ContractCompiler::cloneRuntime() const
 	a << u256(0x20) << u256(0) << Instruction::RETURN;
 	return make_shared<eth::Assembly>(a);
 }
+
+void ContractCompiler::addScopedVariable(VariableDeclaration const& _decl)
+{
+	m_scopedVariables[m_modifierDepth][_decl.scope()].push_back(&_decl);
+	if (stackSizeOfCurrentLocalVariables() > 17)
```

Comment:
Do we really need this here? It might be that a variable is added that is never accessed.

Reply:
Or actually something like this did not compile in the old version:

```
{
 unit a1; 
 a1 = 7;
 uint a2; ... uint a15; uint a16;
 { uint a17 = f(); a16 = a17; }
}
```
but it will with in the current version.
```

Reply:
That compiles, but adding another unused variable breaks compilation.
If we remove the `if` you mentioned, where is this error caught later on?

Reply:
The error should be caught at the point where the value of the variable is requested or a new value is to be stored in the variable. Note that the limit is actually smaller inside more complex expressions, so this check here would not suffice anyway.

Reply:
Ok, gonna remove it